packages/design-system/src/components/N8nPillList/PillList.vue

```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts" generic="Value, Item extends { name: string; initialValue: Value }">
+import { computed } from 'vue';
+
+defineSlots<{
+	displayItem: (props: Item) => unknown;
+}>();
+
+type PillListProps = {
+	inputs: Item[];
+	disabled?: boolean;
+};
+
+const props = withDefaults(defineProps<PillListProps>(), {
+	inputs: () => [],
+	disabled: false,
+});
+
+// Record<inputs[k].name, initialValue>
+// Note that only the keys will stay up to date to reflect selected keys
+// Whereas the values will not automatically update if the related slot value is updated
+const selectedItems = defineModel<Record<string, Value>>({ required: true });
+
+const inputMap = computed(() => Object.fromEntries(props.inputs.map((x) => [x.name, x] as const)));
+
+const visiblePills = computed(() => {
+	return props.inputs.filter((pill) => !selectedItems.value.hasOwnProperty(pill.name));
+});
+
+const sortedSelectedItems = computed(() => {
+	return [
+		...Object.entries(selectedItems.value).map(([name, initialValue]) => ({
+			...inputMap.value[name],
+			initialValue,
+		})),
+	].sort((a, b) => (a.name[0] < b.name[0] ? -1 : 1));
```

Comment:
`a.name.localeCompare(b.name)`, you could also extract and reuse this on line 51